sound/soc/sof/loader.c

```diff
@@ -129,6 +130,70 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+/* parse extended FW manifest data structures */
+static int snd_sof_fw_xman_parse(struct snd_sof_dev *sdev, const void *data)
+{
+	struct xman_header *head = (struct xman_header *)data;
+	struct xman_elem_header *elem_hdr;
+	u32 remaining = head->size - head->header_size;
+	uintptr_t iptr;
+	int ret = 0;
+
+	if (XMAN_VERSION_INCOMPATIBLE(XMAN_VERSION, head->version))
+		dev_warn(sdev->dev, "warning: extended manifest version 0x%X differ from used 0x%X\n",
+			 head->version, XMAN_VERSION);
+
+	/* get first header */
+	iptr = (uintptr_t)data + head->header_size;
+	elem_hdr = (struct xman_elem_header *)iptr;
+
+	while (remaining > 0) {
+		dev_dbg(sdev->dev, "found ext header type %d size 0x%x\n",
+			elem_hdr->type, elem_hdr->size);
+
+		/* process structure data */
+		switch (elem_hdr->type) {
+		default:
+			dev_warn(sdev->dev, "warning: unknown ext header type %d size 0x%x\n",
+				 elem_hdr->type, elem_hdr->size);
+			ret = 0;
```

Comment:
is this necessary? ret is already initiliazed to 0?

Reply:
ret = -EINVAL?

Reply:
Yes, it is little redundant, so I removed `ret` assign during declaration and keep at this line, to keep situation descriptive - after parsing unknown header, parser should omit him and keep working, then mechanism will be easy extensible with backward compatibility.